### PR TITLE
Merge upstream

### PR DESCRIPTION
**Pull request by sonnyp**
# WAT

Merge upstream master
Deletes our commonjs/ie9 fixes are it was fixed upstream

tested and updated on frontend here: https://github.com/teambox/teambox-frontend/pull/2568
# WHY

Several WebSocket/BOSH fixes, no need for patches anymore

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,38 @@
...
+* Rerelease of 1.2.0 but with a semver tag and proper formatting of bower.json
+  for usage with Bower.io.
+
+## Version 1.2.0 - 2015-02-21
+* Add bower package manager support.
+* Add commandline testing support via qunit-phantomjs-runner
+* Add integrated testing via TravisCI.
+* Fix Websocket connections now use the current XMPP-over-WebSockets RFC
+* #25 Item-not-found-error caused by long term request.
+* #29 Add support for the Asynchronous Module Definition (AMD) and require.js
```
Comment (masylum) [inline CHANGELOG.md:22]: finally

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,38 @@
 # Strophe.js Change Log
 
+## Version 1.2.2 - 2015-06-20
+* #109 Explicitly define AMD modules to prevent errors with AlmondJS and AngularJS.
+* #111 Fixed IE9 compatibility.
```
Comment (masylum) [inline CHANGELOG.md:5]: LOL

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,38 @@
...
+
+## Version 1.2.1 - 2015-02-22 
+* Rerelease of 1.2.0 but with a semver tag and proper formatting of bower.json
+  for usage with Bower.io.
+
+## Version 1.2.0 - 2015-02-21
+* Add bower package manager support.
+* Add commandline testing support via qunit-phantomjs-runner
+* Add integrated testing via TravisCI.
+* Fix Websocket connections now use the current XMPP-over-WebSockets RFC
```
Comment (sonnyp) [inline CHANGELOG.md:20]: :+1: :+1:

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,38 @@
...
+* #25 Item-not-found-error caused by long term request.
+* #29 Add support for the Asynchronous Module Definition (AMD) and require.js
+* #30 Base64 encoding problem in some older browsers.
+* #45 Move xhlr plugin to strophejs-plugins repo.
+* #60 Fixed deletion of handlers in websocket connections
+* #62 Add `xmlunescape` method.
+* #67 Use correct Content-Type in BOSH
+* #70 `_onDisconnectTimeout` never tiggers because maxRetries is undefined.
```
Comment (sonnyp) [inline CHANGELOG.md:28]: :+1: :+1:

**Pull request merged**